modules/azure/client_factory.go

```diff
@@ -967,3 +1002,53 @@ func getBaseURI() (string, error) {
 	}
 	return baseURI, nil
 }
+
+// getArmAppContainersClientFactory gets an arm app containers client factory
+func getArmAppContainersClientFactory(subscriptionID string) (*armappcontainers.ClientFactory, error) {
+	targetSubscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	clientCloudConfig, err := getClientCloudConfig()
+	if err != nil {
+		return nil, err
+	}
+	cred, err := azidentity.NewDefaultAzureCredential(&azidentity.DefaultAzureCredentialOptions{
+		ClientOptions: azcore.ClientOptions{
+			Cloud: clientCloudConfig,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return armappcontainers.NewClientFactory(targetSubscriptionID, cred, &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Cloud: clientCloudConfig,
+		},
+	})
+}
+
+func getClientCloudConfig() (cloud.Configuration, error) {
+	envName := getDefaultEnvironmentName()
+	switch strings.ToUpper(envName) {
```

Comment:
What if the envName doesn't match any known environment, it returns an error. You could consider logging the defaulting behavior or providing guidance on what values are expected.

Reply:
I could find any good examples on how logging is done without having the `testing.T` at hand with `terratest.modules.logger`. Do you have any preferences or is it sufficient to just add enough context to the error returned?

Reply:
Hmm can you just pass in the testing.T variable along and share the context that weay?